api/pkg/db/migration/202107291608_refresh_all_tables.go

```diff
@@ -0,0 +1,262 @@
+	"gorm.io/gorm"
+)
+
+func refreshAllTables(log *log.Logger) *gormigrate.Migration {
+	return &gormigrate.Migration{
+		ID: "202107291608_refresh_all_tables",
+		Migrate: func(db *gorm.DB) error {
+			var user []model.User
+			if err := db.Find(&user).Error; err != nil {
+				log.Error(err)
+				return err
+			}
+
+			var user_scopes []model.UserScope
+			if err := db.Find(&user_scopes).Error; err != nil {
+				log.Error(err)
+				return err
+			}
+
+			var user_rating []model.UserResourceRating
+			if err := db.Find(&user_rating).Error; err != nil {
+				log.Error(err)
+				return err
+			}
+
+			var tags []model.Tag
+			if err := db.Find(&tags).Error; err != nil {
+				log.Error(err)
+				return err
+			}
+
+			var sync_job []model.SyncJob
+			if err := db.Find(&sync_job).Error; err != nil {
+				log.Error(err)
+				return err
+			}
+
+			var scopes []model.Scope
+			if err := db.Find(&scopes).Error; err != nil {
+				log.Error(err)
+				return err
+			}
+
+			var resources []model.Resource
+			if err := db.Find(&resources).Error; err != nil {
+				log.Error(err)
+				return err
+			}
+
+			var resource_versions []model.ResourceVersion
+			if err := db.Find(&resource_versions).Error; err != nil {
+				log.Error(err)
+				return err
+			}
+
+			var resource_tags []model.ResourceTag
+			if err := db.Find(&resource_tags).Error; err != nil {
+				log.Error(err)
+				return err
+			}
+
+			var configs []model.Config
+			if err := db.Find(&configs).Error; err != nil {
+				log.Error(err)
+				return err
+			}
+
+			var catalog []model.Catalog
+			if err := db.Find(&catalog).Error; err != nil {
+				log.Error(err)
+				return err
+			}
+
+			var catalog_error []model.CatalogError
+			if err := db.Find(&catalog_error).Error; err != nil {
+				log.Error(err)
+				return err
+			}
+
+			txn := db.Begin()
+			if err := db.Migrator().DropTable(&model.Category{},
+				&model.Tag{},
+				&model.Catalog{},
+				&model.CatalogError{},
+				&model.Resource{},
+				&model.ResourceVersion{},
+				&model.User{},
+				&model.UserResourceRating{},
+				&model.SyncJob{},
+				&model.Scope{},
+				&model.UserScope{},
+				&model.Config{},
+				&model.ResourceTag{},
+			); err != nil {
+				log.Error(err)
+				txn.Rollback()
+				return err
+			}
+
+			if err := db.AutoMigrate(
+				&model.Tag{},
+				&model.Catalog{},
+				&model.CatalogError{},
+				&model.Resource{},
+				&model.ResourceVersion{},
+				&model.User{},
+				&model.UserResourceRating{},
+				&model.SyncJob{},
+				&model.Scope{},
+				&model.UserScope{},
+				&model.Config{},
+				&model.ResourceTag{},
+			); err != nil {
+				log.Error(err)
+				txn.Rollback()
+				return err
+			}
+
+			if len(catalog) > 0 {
+				if err := db.Create(
+					&catalog,
+				).Error; err != nil {
+					log.Error(err)
+					txn.Rollback()
+					return err
+				}
+			}
+
+			if len(catalog_error) > 0 {
+				if err := db.Create(
+					&catalog_error,
+				).Error; err != nil {
+					log.Error(err)
+					txn.Rollback()
```

Comment:
we can move all queries in a func and handle the rollback from there instead of adding for each query